--- tools/zip/BUILD.tools.orig	2021-03-22 20:12:49.000000000 +0100
+++ tools/zip/BUILD.tools	2021-03-22 20:13:03.000000000 +0100
@@ -1,9 +1,14 @@
 package(default_visibility = ["//visibility:public"])
 
+genrule(
+    name = "zipper-exec",
+    outs = ["zipper.bin"],
+    cmd_bash = "ln -s `which zipper` $@",
+    executable = 1,
+    visibility = ["//visibility:public"],
+)
+
 filegroup(
     name = "zipper",
-    srcs = select({
-        "//src/conditions:remote": ["//third_party/ijar:zipper"],
-        "//conditions:default": glob(["zipper/*"]),
-   }),
+    srcs = [":zipper-exec"],
 )
