diff --git a/third_party/grpc/BUILD b/third_party/grpc/BUILD
index 3bd85ba6b3..b006428364 100644
--- a/third_party/grpc/BUILD
+++ b/third_party/grpc/BUILD
@@ -58,16 +58,12 @@ distrib_java_import(
     enable_distributions = ["debian"],
 )

-distrib_cc_binary(
+genrule(
     name = "grpc-java-plugin",
-    enable_distributions = ["debian"],
-    srcs = [
-        "compiler/src/java_plugin/cpp/java_generator.cpp",
-        "compiler/src/java_plugin/cpp/java_generator.h",
-        "compiler/src/java_plugin/cpp/java_plugin.cpp",
-    ],
-    copts = ["-w"],
-    deps = ["//third_party/protobuf:protoc_lib"],
+    outs = ["grpc-java-plugin.out"],
+    cmd = "ln -s ${BUILD_PREFIX}/bin/grpc_java_plugin $@",
+    executable = 1,
+    visibility = ["//visibility:public"],
 )

 alias(
