diff --git a/src/BUILD b/src/BUILD
index d6713d5a8f..60e667f1f0 100644
--- a/src/BUILD
+++ b/src/BUILD
@@ -375,16 +375,41 @@ genrule(
 )
 
 [genrule(
-    name = "bazel-bin" + suffix,
+    name = "bazel-bin-install-name" + suffix,
     srcs = [
         "//src/main/cpp:client",
+    ],
+    outs = ["bazel-install-name" + suffix],
+    cmd = "rm -f $@; cp $(location //src/main/cpp:client) $@; chmod a+rwx $@; ${INSTALL_NAME_TOOL} -rpath ${PREFIX}/lib '@loader_path/../lib/' $@",
+    executable = 1,
+    output_to_bindir = 1,
+    visibility = [
+        "//scripts:__pkg__",  # For bash completion generation
+        "//scripts/packages:__pkg__",  # For installer generation
+        "//src/java:__subpackages__",  # For command line reference generation
+        "//src/test:__subpackages__",  # For integration tests
+    ],
+) for suffix, jdk in [
+    ("-dev", "_jdk_allmodules"),
+    ("", "_jdk_minimal"),
+    ("_jdk_allmodules", "_jdk_allmodules"),
+    ("_jdk_minimal", "_jdk_minimal"),
+    ("_dev_jdk", "_dev_jdk"),
+    ("_notools", "_notools"),
+    ("_nojdk", "_nojdk"),
+]]
+
+[genrule(
+    name = "bazel-bin" + suffix,
+    srcs = [
+        "bazel-bin-install-name" + suffix,
         "package-zip" + jdk,
     ],
     outs = ["bazel" + suffix],
     # In msys, a file path without .exe suffix(say foo), refers to a file with .exe
     # suffix(say foo.exe), if foo.exe exists and foo doesn't. So, on windows, we
     # need to remove bazel.exe first, so that cat to bazel won't fail.
-    cmd = "rm -f $@; cat $(location //src/main/cpp:client) $(location :package-zip" + jdk + ") > $@ && zip -qA $@",
+    cmd = "rm -f $@; cat $(location :bazel-bin-install-name" + suffix + ") $(location :package-zip" + jdk + ") > $@ && zip -qA $@",
     executable = 1,
     output_to_bindir = 1,
     visibility = [

