--- compile.sh  2021-03-24 22:13:32.000000000 +0100
+++ compile.sh  2021-03-24 22:14:51.000000000 +0100
@@ -69,6 +69,7 @@
   --platforms=@local_config_platform//:host \
   || fail "Could not build Bazel"
 bazel_bin_path="$(get_bazel_bin_path)/src/bazel_nojdk${EXE_EXT}"
+bazel_bin_path="${bazel_bin_path//BUILD_CPU/TARGET_CPU}"
 [ -e "$bazel_bin_path" ] \
   || fail "Could not find freshly built Bazel binary at '$bazel_bin_path'"
 cp -f "$bazel_bin_path" "output/bazel${EXE_EXT}" \

