From 47ad67a76213609ad26472e63874ab3769322ae1 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 25 May 2023 20:35:02 +1100
Subject: [PATCH 03/13] use C++17

---
 .bazelrc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/.bazelrc b/.bazelrc
index 89de37ca21..7a95af5f28 100644
--- a/.bazelrc
+++ b/.bazelrc
@@ -92,3 +92,5 @@ test:ci-windows --test_env=JAVA_HOME
 test:ci-windows --test_env=TEST_INSTALL_BASE=C:/b/bazeltest_install_base
 test:ci-windows --test_env=REPOSITORY_CACHE=C:/b/bazeltest_repo_cache
 test:ci-windows --test_tag_filters=-no_windows,-slow
+
+build --cxxopt=-std=c++17 --host_cxxopt=-std=c++17
