From bb898457b46ce71f6020fa46081db9623946b467 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 25 May 2023 20:35:02 +1100
Subject: [PATCH 04/15] use C++17

---
 .bazelrc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/.bazelrc b/.bazelrc
index 7ed38a73bf..9d8d363a5f 100644
--- a/.bazelrc
+++ b/.bazelrc
@@ -48,6 +48,8 @@ build --tool_java_language_version=11
 # Manually enable cc toolchain resolution before it is flipped. https://github.com/bazelbuild/bazel/issues/7260
 build --incompatible_enable_cc_toolchain_resolution
 
+build --cxxopt=-std=c++17 --host_cxxopt=-std=c++17
+
 # User-specific .bazelrc
 try-import %workspace%/user.bazelrc
 
