From 26f301459417fa0a809c78a49a58493e4184153d Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Sat, 12 Feb 2022 22:05:55 +0100
Subject: [PATCH 5/9] grpc-java-plugin from build

---
 third_party/grpc-java/BUILD | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/third_party/grpc-java/BUILD b/third_party/grpc-java/BUILD
index 34cb26c0d1..37b3b077f5 100644
--- a/third_party/grpc-java/BUILD
+++ b/third_party/grpc-java/BUILD
@@ -76,14 +76,10 @@ distrib_java_import(
     ],
 )
 
-distrib_cc_binary(
+genrule(
     name = "grpc-java-plugin",
-    srcs = [
-        "compiler/src/java_plugin/cpp/java_generator.cpp",
-        "compiler/src/java_plugin/cpp/java_generator.h",
-        "compiler/src/java_plugin/cpp/java_plugin.cpp",
-    ],
-    copts = ["-w"],
-    enable_distributions = ["debian"],
-    deps = ["//third_party/protobuf:protoc_lib"],
+    outs = ["grpc-java-plugin.out"],
+    cmd = "ln -s ${BUILD_PREFIX}/bin/grpc_java_plugin $@",
+    executable = 1,
+    visibility = ["//visibility:public"],
 )
