From 7df286df0677df33b69985ebca436b2edc351c5d Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Sat, 12 Feb 2022 22:05:55 +0100
Subject: [PATCH 05/12] grpc-java-plugin from build

---
 third_party/grpc-java/BUILD | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/third_party/grpc-java/BUILD b/third_party/grpc-java/BUILD
index 960e4fa231..34c6caa75a 100644
--- a/third_party/grpc-java/BUILD
+++ b/third_party/grpc-java/BUILD
@@ -64,14 +64,10 @@ filegroup(
     srcs = GRPC_JARS,
 )
 
-distrib_cc_binary(
+genrule(
     name = "grpc-java-plugin",
-    srcs = [
-        "compiler/src/java_plugin/cpp/java_generator.cpp",
-        "compiler/src/java_plugin/cpp/java_generator.h",
-        "compiler/src/java_plugin/cpp/java_plugin.cpp",
-    ],
-    copts = ["-w"],
-    enable_distributions = ["debian"],
-    deps = ["//third_party/protobuf:protoc_lib"],
+    outs = ["grpc-java-plugin.out"],
+    cmd = "ln -s ${BUILD_PREFIX}/bin/grpc_java_plugin $@",
+    executable = 1,
+    visibility = ["//visibility:public"],
 )
