From 0e4cbe4eefe7690c3ae36ad5d8c0ded5c7811dc9 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Sat, 12 Feb 2022 22:08:16 +0100
Subject: [PATCH 08/14] bazel bin loader path

---
 src/BUILD | 29 +++++++++++++++++++++++++++--
 1 file changed, 27 insertions(+), 2 deletions(-)

diff --git a/src/BUILD b/src/BUILD
index e0b9cc9cf4..8e2f6fe197 100644
--- a/src/BUILD
+++ b/src/BUILD
@@ -364,16 +364,41 @@ genrule(
 )
 
 [genrule(
-    name = "bazel-bin" + suffix,
+    name = "bazel-bin-install-name" + suffix,
     srcs = [
         "//src/main/cpp:client",
+    ],
+    outs = ["bazel-install-name" + suffix],
+    cmd = "rm -f $@; cp $(location //src/main/cpp:client) $@; chmod a+rwx $@; ${INSTALL_NAME_TOOL} -rpath ${PREFIX}/lib '@loader_path/../lib/' $@",
+    executable = 1,
+    output_to_bindir = 1,
+    visibility = [
+        "//scripts:__pkg__",  # For bash completion generation
+        "//scripts/packages:__pkg__",  # For installer generation
+        "//src/java:__subpackages__",  # For command line reference generation
+        "//src/test:__subpackages__",  # For integration tests
+    ],
+) for suffix, jdk in [
+    ("-dev", "_jdk_allmodules"),
+    ("", "_jdk_minimal"),
+    ("_jdk_allmodules", "_jdk_allmodules"),
+    ("_jdk_minimal", "_jdk_minimal"),
+    ("_dev_jdk", "_dev_jdk"),
+    ("_notools", "_notools"),
+    ("_nojdk", "_nojdk"),
+]]
+
+[genrule(
+    name = "bazel-bin" + suffix,
+    srcs = [
+        "bazel-bin-install-name" + suffix,
         "package-zip" + jdk,
     ],
     outs = ["bazel" + suffix],
     # In msys, a file path without .exe suffix(say foo), refers to a file with .exe
     # suffix(say foo.exe), if foo.exe exists and foo doesn't. So, on windows, we
     # need to remove bazel.exe first, so that cat to bazel won't fail.
-    cmd = "rm -f $@; cat $(location //src/main/cpp:client) $(location :package-zip" + jdk + ") > $@ && zip -qA $@ && chmod a+x $@",
+    cmd = "rm -f $@; cat $(location :bazel-bin-install-name" + suffix + ") $(location :package-zip" + jdk + ") > $@ && zip -qA $@ && chmod a+x $@",
     executable = 1,
     output_to_bindir = 1,
     visibility = [
