From b6ebbdf8f0c6dd263e604abf2a1a2dde5c45c8a0 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Thu, 27 Jun 2024 14:00:58 +0200
Subject: [PATCH 11/15] Use old-style conditions

---
 third_party/blake3/blake3.BUILD | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/third_party/blake3/blake3.BUILD b/third_party/blake3/blake3.BUILD
index d8e76ece86..a7b6843ef4 100644
--- a/third_party/blake3/blake3.BUILD
+++ b/third_party/blake3/blake3.BUILD
@@ -21,6 +21,21 @@ filegroup(
     visibility = ["//third_party:__pkg__"],
 )
 
+config_setting(
+    name = "linux_aarch64",
+    values = {"cpu": "aarch64"},
+    visibility = ["//visibility:public"],
+)
+
+config_setting(
+    name = "macos_arm64",
+    values = {
+        "apple_platform_type": "macos",
+        "cpu": "darwin_arm64",
+    },
+    visibility = ["//visibility:public"],
+)
+
 cc_library(
     name = "blake3",
     srcs = [
@@ -36,7 +51,7 @@ cc_library(
             "c/blake3_sse2_x86-64_unix.S",
             "c/blake3_sse41_x86-64_unix.S",
         ],
-        "@bazel_tools//src/conditions:linux_aarch64": [
+        ":linux_aarch64": [
             "c/blake3_neon.c",
         ],
         "@bazel_tools//src/conditions:windows_x64": [
@@ -48,7 +63,7 @@ cc_library(
         "@bazel_tools//src/conditions:windows_arm64": [
             "c/blake3_neon.c",
         ],
-        "@bazel_tools//src/conditions:darwin_arm64": [
+	":macos_arm64": [
             "c/blake3_neon.c",
         ],
         "//conditions:default": [],
