From 5540c1103ea2dddb408761b7a93b53f6b19a1278 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Fri, 8 Nov 2024 22:04:59 +0100
Subject: [PATCH 12/12] Add missing absl dependency

---
 src/main/cpp/BUILD | 29 +++++++++++++++++++++++++++--
 1 file changed, 27 insertions(+), 2 deletions(-)

diff --git a/src/main/cpp/BUILD b/src/main/cpp/BUILD
index dd3bc40397..b8fac89319 100644
--- a/src/main/cpp/BUILD
+++ b/src/main/cpp/BUILD
@@ -90,6 +90,15 @@ cc_library(
     ],
 )
 
+# cannot do `[...] + select(...)` in starlark, so factor out common libraries into
+# a variable; note that this is likely not the full list, because we're inheriting
+# the linkopts from protobuf (see third_party/systemlibs/protobuf/BUILD), so several
+# abseil libraries will already be present
+_CLIENT_LINKOPTS = [
+    "-labsl_log_severity",
+    "-labsl_synchronization",
+]
+
 cc_binary(
     name = "client",
     srcs = [
@@ -107,7 +116,7 @@ cc_binary(
         "//conditions:default": ["-Wno-sign-compare"],
     }),
     linkopts = select({
-        "//src/conditions:darwin": [
+        "//src/conditions:darwin": _CLIENT_LINKOPTS + [
         ],
         "//src/conditions:freebsd": [
             "-lprocstat",
@@ -117,7 +126,7 @@ cc_binary(
         ],
         "//src/conditions:windows": [
         ],
-        "//conditions:default": [
+        "//conditions:default": _CLIENT_LINKOPTS + [
             "-lrt",
             "-ldl",
         ],
@@ -138,6 +147,22 @@ cc_binary(
         "//src/main/protobuf:command_server_cc_grpc",
         "//src/main/protobuf:command_server_cc_proto",
         "//third_party/ijar:zip",
+        # required dependencies we need to add because we're unvendoring; however, any
+        # `@abseil-cpp//absl/foo` added here don't actually take effect from the POV
+        # of the linker, because abseil will by default look for a static library, e.g.
+        #   bazel-out/darwin-opt/bin/external/abseil-cpp~/absl/foo/libfoo.a
+        # This could be because we're building abseil itself with CMake rather than
+        # bazel (so the metadata might be off). Rather than come up with complicated
+        # wrappers, we abuse linkopts to add the right libraries (see above).
+        "@abseil-cpp//absl/base:log_severity",
+        "@abseil-cpp//absl/log:absl_check",
+        "@abseil-cpp//absl/container:flat_hash_map",
+        "@abseil-cpp//absl/container:btree",
+        "@abseil-cpp//absl/log:log",
+        "@abseil-cpp//absl/log:absl_log",
+        "@abseil-cpp//absl/strings:cord",
+        "@abseil-cpp//absl/status:statusor",
+        "@abseil-cpp//absl/utility:if_constexpr",
     ],
 )
 
