From 8aae4c88aef2ebfc8f3a74ee153cd84a978f5dc5 Mon Sep 17 00:00:00 2001
From: Uwe Korn <uwe.korn@quantco.com>
Date: Thu, 27 Jun 2024 12:36:53 +0000
Subject: [PATCH] Use old-sytle condition for linux64

---
 third_party/blake3/blake3.BUILD | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/third_party/blake3/blake3.BUILD b/third_party/blake3/blake3.BUILD
index a7b6843..d237ebb 100755
--- a/third_party/blake3/blake3.BUILD
+++ b/third_party/blake3/blake3.BUILD
@@ -21,6 +21,12 @@ filegroup(
     visibility = ["//third_party:__pkg__"],
 )
 
+config_setting(
+    name = "linux_x86_64",
+    values = {"cpu": "k8"},
+    visibility = ["//visibility:public"],
+)
+
 config_setting(
     name = "linux_aarch64",
     values = {"cpu": "aarch64"},
@@ -43,7 +49,7 @@ cc_library(
         "c/blake3_dispatch.c",
         "c/blake3_portable.c",
     ] + select({
-        "@bazel_tools//src/conditions:linux_x86_64": [
+        ":linux_x86_64": [
             "c/blake3_avx2_x86-64_unix.S",
             # Disable to appease bazel-ci which uses ubuntu-18 (EOL) and GCC 7
             # lacking the headers to compile AVX512.
